File: backend/furniture/src/main/java/com/khumaloterrance/furniture/entity/ItemToPurchase.java
package com.khumaloterrance.furniture.entity;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;


@Entity
@Getter
@Setter
@ToString
public class ItemToPurchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long productId;

    private int quantity;

    private String imageUrl;

    private int unitPrice;

    @ManyToOne
    @JoinColumn(name = "orders_id")
    private Order order;
}

File: frontend/src/app/common/item-to-purchase.ts
import { CartItem } from "./cartItem";

export class ItemToPurchase {

    imageUrl: string | undefined;
    quantity: number | undefined;
    unitPrice: number | undefined;
    productId: number | undefined;

    constructor(cartItem: CartItem){
        this.imageUrl = cartItem.imageUrl;
        this.quantity = cartItem.quantity;
        this.unitPrice = cartItem.unitPrice;
        this.productId = cartItem.id;
    }
}
